doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,237 +1257,211 @@ Managing the local context
+           move y after z.  (*    x Hx z y Hy   (z was left of y, intuitive case) *)
            Undo.
-           move x before H0.
+           move z after y.  (*    x Hx z y Hy   (z was right of y, see Note above) *)
            Undo.
-           move H0 after H.
+           move x after Hy.  (*   y z Hy x Hx   (Hx depends on x, so moved) *)
            Undo.
-           move H0 before H.
-
-.. tacn:: rename @ident__1 into @ident__2
-   :name: rename
-
-   This renames hypothesis :n:`@ident__1` into :n:`@ident__2` in the current
-   context. The name of the hypothesis in the proof-term, however, is left
-   unchanged.
-
-   .. tacv:: rename {+, @ident__i into @ident__j}
+           move x before Hy.  (*  y z x Hx Hy *)
+           Undo.
+           move Hy after Hx.  (*  x y Hy Hx z *)
+           Undo.
+           move Hy before Hx.  (* x Hx y Hy z *)
 
-      This renames the variables :n:`@ident__i` into :n:`@ident__j` in parallel.
-      In particular, the target identifiers may contain identifiers that exist in
-      the source context, as long as the latter are also renamed by the same
-      tactic.
+.. tacn:: rename {+, @ident__1 into @ident__2 }
 
-   .. exn:: No such hypothesis.
-      :undocumented:
+   Renames hypothesis :n:`@ident__1` into :n:`@ident__2` for each pair of :n:`@ident`\s.
+   Renaming is done simultaneously, which permits swapping the names of 2 hypotheses.
+   (Note that the renaming is applied in the context and the existential
+   variables, but the proof term doesn't change.)
 
    .. exn:: @ident is already used.
       :undocumented:
 
-.. tacn:: set (@ident := @term)
-   :name: set
+.. tacn:: set @alias_definition {? @occurrences }
+          set @one_term {? @as_name } {? @occurrences }
+   :name: set; _
 
-   This replaces :token:`term` by :token:`ident` in the conclusion of the
-   current goal and adds the new definition :n:`@ident := @term` to the
-   local context.
-
-   If :token:`term` has holes (i.e. subexpressions of the form “`_`”), the
-   tactic first checks that all subterms matching the pattern are compatible
-   before doing the replacement using the leftmost subterm matching the
-   pattern.
-
-   .. exn:: The variable @ident is already defined.
-      :undocumented:
+   .. insertprodn alias_definition as_name
 
-   .. tacv:: set (@ident := @term) in @goal_occurrences
+   .. prodn::
+      alias_definition ::= ( @ident {* @simple_binder } := @term )
+      simple_binder ::= @name
+      | ( {+ @name } : @term )
+      as_name ::= as @ident
 
-      This notation allows specifying which occurrences of :token:`term` have
-      to be substituted in the context. The :n:`in @goal_occurrences` clause
-      is an occurrence clause whose syntax and behavior are described in
-      :ref:`goal occurrences <occurrenceclauses>`.
+   The first form adds a new local definition
+   :n:`@ident := fun {* @simple_binder } => @term` and replaces :n:`@term` in
+   the goal with a new variable :n:`@ident`.
```

Comment:
Maybe worth starting with the case with no simple_binder so that readers don't think that a `fun` will be introduced in this case?